src/io/flutter/inspector/FlutterWidget.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2017 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.inspector;
+
+import com.google.gson.*;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A representation of a Flutter widget.
+ */
+public class FlutterWidget {
+
+  private static final Catalog catalog = new Catalog();
+  private final JsonObject json;
+
+  private FlutterWidget(@NotNull JsonObject json) {
+    this.json = json;
+  }
+
+  @Contract(pure = true)
+  @NotNull
+  public static Catalog getCatalog() {
+    return catalog;
+  }
+
+  @Nullable
+  public String getName() {
+    return getStringMember("name");
+  }
+
+  @Nullable
+  public List<String> getCategories() {
+    return getValues("categories");
+  }
+
+  @Nullable
+  public List<String> getSubCategories() {
+    return getValues("subcategories");
+  }
+
+  @NotNull
+  private List<String> getValues(@NotNull String member) {
+    if (!json.has(member)) {
+      return Collections.emptyList();
+    }
+
+    final JsonArray rawValues = json.getAsJsonArray(member);
+    final ArrayList<String> values = new ArrayList<>(rawValues.size());
+    rawValues.forEach(element -> values.add(element.getAsString()));
+
+    return values;
+  }
+
+  @Nullable
+  private String getStringMember(String memberName) {
+    if (!json.has(memberName)) return null;
```

Comment:
this looks familiar. Maybe add it as a static method in a util class as we seem to be using it in a bunch of places?

Reply:
Sure.  Done!